logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/ObserveCertificateForCurrentClientUseCase.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.e2ei.usecase
+
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.data.e2ei.E2EIRepository
+import com.wire.kalium.logic.functional.map
+import kotlinx.datetime.Clock
+
+/**
+ * Use case to observe current client certificate and check if it is expired.
+ */
+interface ObserveCertificateForCurrentClientUseCase {
+    suspend operator fun invoke()
+}
+
+internal class ObserveCertificateForCurrentClientUseCaseImpl(
```

Comment:
question: why we need this useCase in this PR?

Reply:
Because we need to keep checking the CRL when the certificate is not expired.